### PR TITLE
fix unbound type parameter

### PR DESCRIPTION
**Pull request by nsajko**

### CODE REVIEW

**Comment by jishnub** [comment]
Gentle bump @tkf

**Comment by nsajko** [comment]
bump

**Comment by jishnub** [comment]
Gentle bump @JuliaFolds, could someone merge this?

**Comment by MilesCranmer** [comment]
@MasonProtter @DilumAluthge I think you have merge access, and I'm not sure if @tkf is doing much open-source anymore. Could one of you merge and release?

This unbound type param shows up as a warning during precompilation for all [221+ dependent projects](https://juliahub.com/ui/Packages/MicroCollections/UymYM/0.1.3?page=2).

**Comment by DilumAluthge** [comment]
I unfortunately do not have the commit bit here.

**Comment by MasonProtter** [comment]
I also unfortunately don't have commit rights in this repo.

**Comment by MilesCranmer** [comment]
That is unfortunate. Unless @tkf is able to update permissions, maybe a fork could be made instead?

I think a lot of the dependencies are via Metatheory.jl (=>ThreadsX.jl=>MicroCollections.jl). So one way to fix this warning would be to have forked ThreadsX and Microcollections.jl and introduce separate dependencies. 

@ChrisRackauckas @0x0f0f0f @shashi would you be open to having forks of these under @JuliaSymbolics so they can be under community maintenance?

File: src/onehot.jl [hunk 1]
```diff
@@ -76,8 +76,8 @@ end
 @inline OneHotArray{<:Any,N}(index_value::Pair{<:Any,T}, shape::DimsLike) where {N,T} =
     OneHotArray(index_value, astuple(shape))::OneHotArray{<:Any,N}
 
-@inline OneHotVector{T}(index_value::Pair, length::Integer) where {T,N} =
-    OneHotArray{T}(index_value, (length,))::OneHotArray{T,N}
+@inline OneHotVector{T}(index_value::Pair, length::Integer) where {T} =
+    OneHotArray{T}(index_value, (length,))::OneHotArray{T,1}
```
Comment (MilesCranmer) [inline src/onehot.jl:80]:
```suggestion
    OneHotArray{T}(index_value, (length,))
```
(Not sure why it actually needs the type assertion?)

**Comment by 0x0f0f0f** [comment]
> That is unfortunate. Unless @tkf is able to update permissions, maybe a fork could be made instead?
> 
> I think a lot of the dependencies are via Metatheory.jl (=>ThreadsX.jl=>MicroCollections.jl). So one way to fix this warning would be to have forked ThreadsX and Microcollections.jl and introduce separate dependencies.
> 
> @ChrisRackauckas @0x0f0f0f @shashi would you be open to having forks of these under @JuliaSymbolics so they can be under community maintenance?

ThreadsX Dependency is removed in MT 2.0 - I do not have time nor funding to keep maintaining MT in a state that it's usable by Symbolics - and we weren't able to organize something to keep MT alive - next SymbolicsUtils.jl is sadly deprecating MT.

EDIT - SymbolicUtils v20 is no longer depending on Metatheory.

**Comment by ChrisRackauckas** [comment]
>EDIT - SymbolicUtils v20 is no longer depending on Metatheory.

v1.0, so yeah this was removed from the dependency tree with the Unityper changes.

**Comment by simsurace** [comment]
What is the status of this? Can it be merged?

**Comment by jishnub** [comment]
There's no active member who has merge rights, so this is stalled. Perhaps forking this repo is the only solution.

In case you have the requisite rights for this repo, please proceed with the merge. It'll be a great help.

**Comment by BioTurboNick** [comment]
Who is an owner of JuliaFolds? Can they not control who has rights on the repository?

**Comment by DilumAluthge** [comment]
IIRC @tkf is the only owner. I've messaged him to ask if he can promote me from member to owner, but I haven't heard back.

**Comment by DilumAluthge** [comment]
If MetaTheory has removed the dependency on ThreadsX, hopefully that fixes the issue for most downstream consumers?

**Comment by MilesCranmer** [comment]
It seems as though @tkf hasn't created a commit since last summer (the commits on his profile are merge commits from others based on his older code). His Julia discourse shows his last login was October, 2022. Does anyone know him; is he doing okay? Maybe he started a new position outside of the Julia community?

If you can contact him, maybe you could request whether he could login to GitHub and open up permissions in these key repos to trusted JuliaLang team members?

**Comment by simsurace** [comment]
I don't know him outside of his Julia contributions, but tried to ping him on Slack and by email.

**Comment by simsurace** [comment]
If he cannot contribute anymore, I hope he can assign a successor. JuliaFolds is extremely important for the wider Julia ecosystem.

**Comment by DilumAluthge** [comment]
I have done the following steps:

1. Created a new GitHub organization: https://github.com/JuliaFolds2
2. Added Viral and Stefan as owners of the JuliaFolds2 organization (so that the organization has three owners)
3. Forked MicroCollections.jl to JuliaFolds2: https://github.com/JuliaFolds2/MicroCollections.jl
    - We can fork all the other JuliaFolds packages as well. I just forked this one to get started.
4. Made a PR to General to set JuliaFolds2/MicroCollections.jl as the official repo:
    - https://github.com/JuliaRegistries/General/pull/79127

Is there any volunteer to take this PR and make it again in the https://github.com/JuliaFolds2/MicroCollections.jl repo?

**Comment by jishnub** [comment]
Should the forks be converted to independent repos? From what I remember, there are issues with `CompatHelper` on forks.

Also, is this the general protocol to deal with such situations in the future?

I've also created a new PR in that repo.

**Comment by DilumAluthge** [comment]
> From what I remember, there are issues with `CompatHelper` on forks.

I think I've managed to fix the CompatHelper issue (https://github.com/JuliaRegistries/CompatHelper.jl/pull/446).

**Comment by jeremiahpslewis** [comment]
@DilumAluthge Would you be willing to tag a new release on the fork, now that this fix has been merged there?

**Comment by jishnub** [comment]
IIUC it's merged into the registry already: https://github.com/JuliaRegistries/General/pull/79646

**Comment by jeremiahpslewis** [comment]
@jishnub Missed this. Thanks!

**Pull request closed without merging**